Source/WebKit/UIProcess/ios/WKContentView.mm

```diff
@@ -965,22 +1001,68 @@ - (NSUInteger)_wk_pageCountForPrintFormatter:(_WKWebViewPrintFormatter *)printFo
     }
 
     if (!pageCount)
+        return nil;
+
+    RetainPtr attributes = adoptNS([[_WKPrintFormattingAttributes alloc] initWithPageCount:pageCount frameID:frameID printInfo:printInfo]);
+
+    return attributes;
+}
+
+- (NSUInteger)_wk_pageCountForPrintFormatter:(_WKWebViewPrintFormatter *)printFormatter
+{
+    if (printFormatter._shouldDrawUsingBitmap)
+        [self _waitForDrawToImageCallbackForPrintFormatterIfNeeded:printFormatter];
+    else
+        [self _waitForDrawToPDFCallbackForPrintFormatterIfNeeded:printFormatter];
+
+    RetainPtr attributes = [self _attributesForPrintFormatter:printFormatter];
+    if (!attributes)
         return 0;
 
-    if (isPrintingOnBackgroundThread) {
-        Locker locker { _pendingBackgroundPrintFormattersLock };
+    return [attributes pageCount];
+}
 
-        if (!_pendingBackgroundPrintFormatters)
-            _pendingBackgroundPrintFormatters = adoptNS([[NSMutableSet alloc] init]);
+- (void)_createImage:(_WKPrintFormattingAttributes *)formatterAttributes isPrintingOnBackgroundThread:(BOOL)isPrintingOnBackgroundThread printFormatter:(_WKWebViewPrintFormatter *)printFormatter
```

Comment:
Why does `isPrintingOnBackgroundThread` need to be a parameter? Same question for the method below.